source/grid/tria.cc

```diff
@@ -4047,6 +4043,479 @@ namespace internal
 
 
 
+      template <int dim, int spacedim>
+      static typename Triangulation<dim, spacedim>::DistortedCellList
+      execute_refinement_isotropic(Triangulation<dim, spacedim> &triangulation,
+                                   const bool check_for_distorted_cells)
```

Comment:
As discussed above, I would suggest to have an isotropic and an anisotropic code path. This here is the isotropic code path for refinement of triangles and quadraletrals.